### PR TITLE
Moved hazard maps and mean and quantile functions from the engine into risklib

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1404518. The tests are green: https://ci.openquake.org/job/zdevel_oq-risklib/167

This also fixes https://bugs.launchpad.net/oq-engine/+bug/1329259

### CODE REVIEW

File: openquake/commonlib/calculators/calc.py [hunk 1]
```diff
@@ -137,3 +137,69 @@ def calc_gmfs(oqparam, sitecol):
...
+    ``poe``.
+
+    :param curves:
+        2D array of floats. Each row represents a curve, where the values
+        in the row are the PoEs (Probabilities of Exceedance) corresponding to
+        ``imls``. Each curve corresponds to a geographical location.
+    :param imls:
+        Intensity Measure Levels associated with these hazard ``curves``. Type
+        should be an array-like of floats.
+    :param float poes:
```
Comment (ptormene) [inline openquake/commonlib/calculators/calc.py]: I guess we should remove `float` from here

File: openquake/commonlib/export/hazard.py [hunk 2]
```diff
@@ -225,9 +229,32 @@ def export_hazard_curves_xml(key, export_dir, sitecol, rlz, curves_by_imt,
...
+
+
+@export.add('hazard_stats_csv')
+def export_stats_csv(key, export_dir, fname, sitecol, data_by_imt):
+    """
+    Export the scalar outputs.
+
+    :param key: output_type and export_type
+    :param export_dir: the directory where to export
+    :param sitecol: site collection
```
Comment (ptormene) [inline openquake/commonlib/export/hazard.py]: missing `:param fname:`

File: openquake/commonlib/tests/calculators/__init__.py [hunk 3]
```diff
@@ -36,21 +36,22 @@ def get_calc(self, testfile, job_ini):
         """
         self.testdir = os.path.dirname(testfile)
         inis = [os.path.join(self.testdir, ini) for ini in job_ini.split(',')]
-        oq = self.oqparam = readinput.get_oqparam(inis)
-        oq.concurrent_tasks = executor._max_workers
+        oq = readinput.get_oqparam(inis)
+        oq.concurrent_tasks = executor._max_workers * 4
```
Comment (ptormene) [inline openquake/commonlib/tests/calculators/__init__.py]: Perhaps this `4` might be transformed into a parameter (or you might add a comment)

File: openquake/commonlib/writers.py [hunk 4]
```diff
@@ -150,3 +160,13 @@ def tostring(node, indent=4):
     writer = StreamingXMLWriter(out, indent)
     writer.serialize(node)
     return out.getvalue()
+
+
+def save_csv(dest, header_rows, sep=',', fmt='%12.8E'):
+    """
+    :param dest: destination filename
+    :param header_rows: header + rows to save
```
Comment (ptormene) [inline openquake/commonlib/writers.py]: You might add the description of `sep` and `fmt`

**Comment by ptormene** [comment]
Apart from the minor comments, LGTM

File: openquake/commonlib/calculators/calc.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/calculators/calc.py]: Done

File: openquake/commonlib/export/hazard.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/export/hazard.py]: Added

File: openquake/commonlib/tests/calculators/__init__.py (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/tests/calculators/__init__.py]: Added a comment

File: openquake/commonlib/writers.py (hunk 4 above)
Comment (micheles) [inline openquake/commonlib/writers.py]: Added

**Pull request merged**